src/app/modules/main/module.nim

```diff
@@ -1308,21 +1308,23 @@ method onStatusUrlRequested*[T](self: Module[T], action: StatusUrlAction, commun
       self.setActiveSection(item)
 
   elif(action == StatusUrlAction.OpenCommunityChannel):
-    var found = false
-    for cId, cModule in self.channelGroupModules.pairs:
-      if(cId == singletonInstance.userProfile.getPubKey()):
-        continue
-      if(cModule.doesCatOrChatExist(chatId)):
-        let item = self.view.model().getItemById(cId)
-        self.setActiveSection(item)
-        cModule.makeChatWithIdActive(chatId)
-        found = true
-        break
-    if not found:
+    let chatId = communityId & channelId
+    if not self.channelGroupModules.hasKey(communityId):
       let communityIdToSpectate = getCommunityIdFromFullChatId(chatId)
       # request community info and then spectate
       self.statusUrlCommunityToSpectate = communityIdToSpectate
       self.communitiesModule.requestCommunityInfo(communityIdToSpectate, importing = false)
+      return
+
+    let cModule = self.channelGroupModules[communityId]
+    let item = self.view.model().getItemById(communityId)
```

Comment:
can we add a check here is community presented in the model

Reply:
hmmm, we have a check above for `if not self.channelGroupModules.hasKey(communityId)`, I hoped for that one. But probably you're right, we should check the `item.isEmpty` as well.

Reply:
refactored this part to use `switchTo`